src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ThrottlingRerouteSpec.groovy

```diff
@@ -50,55 +59,54 @@ class ThrottlingRerouteSpec extends HealthCheckSpecification {
         def differenceInMillis = flowHelper.convertStringTimestampIsoToLong(rerouteTimestamp) - rerouteTriggersEnd
         // reroute starts not earlier than the expected reroute delay
         assert differenceInMillis > (rerouteDelay) * 1000
         // reroute starts not later than 2 seconds later than the expected delay
-        assert differenceInMillis < (rerouteDelay + 2) * 1000
+        assert differenceInMillis < (rerouteDelay + 2.5) * 1000
 
         and: "The oldest broken flow is rerouted when the rerouteDelay runs out"
         def untilReroutesBegin = { rerouteTriggersEnd + rerouteDelay * 1000 - new Date().time }
         def waitTime = untilReroutesBegin() / 1000.0 + PATH_INSTALLATION_TIME * 2
         Wrappers.wait(waitTime) {
-            //Flow should go DOWN or change path on reroute. In our case it doesn't matter which of these happen.
-            assert (northboundV2.getFlowStatus(flows.first().flowId).status == FlowState.DOWN &&
-                    flowHelper.getHistoryEntriesByAction(flows.first().flowId, REROUTE_ACTION).find {
-                        it.taskId =~ (/.+ : retry #1/) })||
-                    northbound.getFlowPath(flows.first().flowId) != flowPaths.first()
+            // Flow should go DOWN or change path on reroute.
+            // In our case it doesn't matter which of these happen.
+            def flow1 = flows.first()
+            assert (flow1.retrieveFlowStatus().status == FlowState.DOWN &&
+                    flow1.retrieveFlowHistory().entries.findAll {it.getAction() == REROUTE_ACTION}
+                            .find { it.taskId =~ (/.+ : retry #1/) })||
+                    flow1.retrieveAllEntityPaths().getMainPath() != flowPaths.first()
         }
 
         and: "The rest of the flows are rerouted too"
         Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
-            flowPaths[1..-1].each { flowPath ->
-                assert (northboundV2.getFlowStatus(flowPath.id).status == FlowState.DOWN &&
-                        flowHelper.getHistoryEntriesByAction(flowPath.id, REROUTE_ACTION).find {
-                    it.taskId =~ (/.+ : retry #1/)
-                })  ||
-                        (northbound.getFlowPath(flowPath.id) != flowPath &&
-                                northboundV2.getFlowStatus(flowPath.id).status == FlowState.UP)
+            flows.subList(1, flows.size()).eachWithIndex { flow, i ->
+                def flowPathIndex = i + 1 // since we start from the flows[1:-1]
+                def currentFlowStatus = flow.retrieveFlowStatus().status
+                assert (currentFlowStatus == FlowState.DOWN &&
+                        flow.retrieveFlowHistory().entries.findAll {it.getAction() == REROUTE_ACTION}
+                                .find { it.taskId =~ (/.+ : retry #1/)}) ||
+                        (flow.retrieveAllEntityPaths().getMainPath() != flowPaths[flowPathIndex] &&
```

Comment:
let's use flowId to get appropriate flow path `flowPaths.find{ it.flowId == flow.flowId} `

Reply:
Done